src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/subquery/SubqueryRewriter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.parser.subquery;
+
+import com.alibaba.druid.sql.dialect.mysql.ast.statement.MySqlSelectQueryBlock;
+
+/**
+ * Rewrite the Predicate Subquery to JOIN.
```

Comment:
I think the documentation should say "rewrites the query to replace the subquery with a join". It does not re-write just the subquery part.

Reply:
Agree. Done.